.github/actions/pre_make/action.yml

```diff
@@ -0,0 +1,44 @@
+name: "Pre-Make"
+runs:
+  using: composite
+  steps:
+    - name: Restore make from cache
+      id: cache-make
+      uses: actions/cache@v3
+      with:
+        path: make.bin
+        key: v13_make.bin-${{ hashFiles('**/make.md5') }}
+    - name: Install make
+      if: steps.cache-make.outputs.cache-hit != 'true'
+      run: |
+        if [ -s make.bin ] ; then
+          echo "make.bin exists; not installing"
+        else
+          echo "/usr/bin/make doesn't exist; installing"
+          sudo apt-get update
```

Comment:
nit (that probably applies in a bunch of other places in this PR):

We run `sudo apt-get update` in a bunch of places when running it ~2 times total should suffice (once at the beginning and once after adding the protobuf ppa). We should probably move this to its own stage that happens before anything else and remove it everywhere else.